spec/unit/puppetfile_spec.rb

```diff
@@ -10,14 +10,35 @@
     )
   end
 
-  describe "a custom puppetfile Puppetfile.r10k" do
+  describe "a custom puppetfile_name" do
     it "is the basedir joined with '/Puppetfile.r10k' path" do
       expect(subject.puppetfile_path).to eq '/some/nonexistent/basedir/Puppetfile.r10k'
     end
   end
 
 end
 
+describe R10K::Puppetfile do
+
+  relative_subject = described_class.new('/some/nonexistent/basedir',
```

Comment:
I think these subjects should go in the `it` blocks, right?

Reply:
I copied this from the test above, which has the `subject` outside of the `it` block. Let me know if you still think I should move them.

Reply:
The above tests use the `subject` RSpec DSL method, see: https://relishapp.com/rspec/rspec-core/docs/subject/explicit-subject .

These appear to be local variables that for some reason are still be in scope when the it block is evaluated.

Reply:
yes, I know. I assume you're saying "yes, change it"?

Reply:
updated

Reply:
Yes.